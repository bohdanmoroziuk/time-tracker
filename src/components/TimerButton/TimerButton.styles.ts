import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    marginTop: 10,
    minWidth: 100,
    borderWidth: 2,
    borderRadius: 3,
  },
  small: {
    fontSize: 14,
    padding: 5,
  },
  large: {
    fontSize: 16,
    padding: 10,
  },
  text: {
    textAlign: 'center',
    fontWeight: 'bold',
  },
  title: {
    fontSize: 14,
    fontWeight: 'bold',
  },
  elapsedTime: {
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
    paddingVertical: 10,
  },
});
