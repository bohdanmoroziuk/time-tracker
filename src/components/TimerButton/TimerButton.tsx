import { FunctionComponent } from 'react';
import { TouchableOpacity, TouchableOpacityProps, Text, ColorValue } from 'react-native';

import styles from './TimerButton.styles';

export interface TimerButtonProps {
  color: ColorValue;
  title: string;
  small?: boolean;
  onPress: TouchableOpacityProps['onPress'];
}

const TimerButton: FunctionComponent<TimerButtonProps> = ({
  color,
  title,
  small = false,
  onPress,
}) => (
  <TouchableOpacity
    style={[styles.container, { borderColor: color }]}
    onPress={onPress}
  >
    <Text
      style={[
        styles.text,
        small ? styles.small : styles.large,
        { color }
      ]}
    >
      {title}
    </Text>
  </TouchableOpacity>
);

export default TimerButton;
