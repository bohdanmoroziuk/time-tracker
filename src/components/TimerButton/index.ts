import TimerButton from './TimerButton';

export type { TimerButtonProps } from './TimerButton';
export default TimerButton;
