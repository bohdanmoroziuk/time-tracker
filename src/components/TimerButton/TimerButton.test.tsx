import { render, fireEvent } from '@testing-library/react-native';

import TimerButton from './TimerButton';

describe('TimerButton', () => {
  const color = "blue";
  const title = "Edit";
  const small = true;
  const mockOnPress = jest.fn();

  it('renders successfully', () => {
    const { toJSON } = render(
      <TimerButton
        color={color}
        title={title}
        small={small}
        onPress={mockOnPress}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('calls an onPress callback when pressed', () => {
    const { getByText } = render(
      <TimerButton
        color={color}
        title={title}
        small={small}
        onPress={mockOnPress}
      />
    );

    fireEvent.press(getByText(title));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
