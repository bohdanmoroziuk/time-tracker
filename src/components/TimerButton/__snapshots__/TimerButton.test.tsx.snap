// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerButton renders successfully 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "blue",
      "borderRadius": 3,
      "borderWidth": 2,
      "marginTop": 10,
      "minWidth": 100,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 14,
          "padding": 5,
        },
        Object {
          "color": "blue",
        },
      ]
    }
  >
    Edit
  </Text>
</View>
`;
